File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {

        // BRUTE FORCE APPROACH------
    //    int maxSum=Integer.MIN_VALUE;
    //    for(int st=0;st<nums.length;st++){
    //     int currentSum=0;
    //     for(int end=st; end<nums.length;end++){
    //         currentSum+= nums[end];
    //         maxSum= Math.max(currentSum, maxSum)
    //     }
    //    }

        // KADANE'S ALGORITHM
        int cs=0;
        int maxSum=Integer.MIN_VALUE;
        for(int i=0;i<nums.length;i++){
             cs=cs+nums[i];
            maxSum= Math.max(cs, maxSum);
            if(cs<0){
                cs=0;
            }
        }
        return maxSum;
    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
   int maxSubArray(vector<int>& nums) {
        int n = nums.size();
        int currentSum = 0;
        int maxSum = INT_MIN;

        for (int i = 0; i < n; i++) {
            currentSum += nums[i];
            maxSum = max(currentSum, maxSum);

            if (currentSum < 0) {
                currentSum = 0;
            }
        }

        return maxSum; 
    }
};